app/reducer.js

```diff
@@ -1,22 +1,20 @@
-import { Map, List } from 'immutable'
+import { Map, Set } from 'immutable'
 import * as actionTypes from './constants'
 import createStore from '../storeCreator'
 
 const initialState = Map({
-  subjects: List()
+  subjects: Set(),
 })
 
 function trackNewSubject(state, action) {
-  const subjects = state.get('subjects')
-  if (subjects.some(s => s.get('name') === action.name))
-    return state
-
-  const subject = Map({ name: action.name, tweets: List(), visibleCount: 50 })
-  return state.merge({
-    subjects: subjects.push(subject)
+  const subject = Map({
+    name: action.name,
+    tweets: Set(),
+    visibleCount: 50,
   })
+  return state.set('subjects', state.get('subjects').add(subject));
```

Comment:
This won't work correctly for the reason we discussed before. It will only guarantee uniqueness as long as `tweets` is empty and `visibleCount` is still 50. As soon as either of those things changes, this will not prevent us from tracking the same subject name twice.